mesonbuild/coredata.py

```diff
@@ -457,10 +458,17 @@ def set_buildtype_from_others(self):
             mode = 'custom'
         self.builtins['buildtype'].set_value(mode)
 
+    def get_all_compiler_options(self):
+        # TODO think about cross and command-line interface.
+        yield self.compiler_options.build
+        #yield self.compiler_options.host
+        #yield self.compiler_options.target
+
     def _get_all_nonbuiltin_options(self):
         yield self.backend_options
         yield self.user_options
-        yield self.compiler_options
+        for co in self.get_all_compiler_options():
+            yield co
```

Comment:
can we just do `yield from self.get_all_compiler_options()`?

Reply:
Oh nice I didn't know about that.